src/synapse/azext_synapse/_params.py

```diff
@@ -3,159 +3,204 @@
+                   help='Additional files used for reference in the main definition file.')
+        c.argument('configuration', type=get_json_object, help='The configuration of Spark session.')
+        c.argument('executors', help='The number of executors.')
+        c.argument('executor_size', arg_type=get_enum_type(['Small', 'Medium', 'Large']), help='The executor size')
+        c.argument('tags', arg_type=tags_type)
 
     for scope in ['show', 'cancel', 'reset-timeout']:
         with self.argument_context('synapse spark session ' + scope) as c:
-            c.argument('session_id', options_list=['--id'], arg_group='Spark Session',
-                       help='The id of the spark session job.')
-
-    with self.argument_context('synapse spark session show') as c:
-        c.argument('detailed', action='store_true',
-                   help='Optional query parameter specifying whether detailed response is returned beyond plain livy.')
+            c.argument('session_id', options_list=['--livy-id'], arg_group='Spark Session',
+                       help='The id of the Spark session job.')
 
-    with self.argument_context('synapse spark session-statement') as c:
-        c.argument('session_id', help='The id of spark session job.')
+    with self.argument_context('synapse spark statement') as c:
+        c.argument('session_id', help='The id of Spark session.')
 
     for scope in ['show', 'cancel']:
-        with self.argument_context('synapse spark session-statement ' + scope) as c:
-            c.argument('statement_id', options_list=['--id'], arg_group="Spark Session-statement",
+        with self.argument_context('synapse spark statement ' + scope) as c:
+            c.argument('statement_id', options_list=['--livy-id'], arg_group="Spark statement",
                        help='The id of the statement.')
 
-    with self.argument_context('synapse spark session-statement create') as c:
-        c.argument('code', help='The code of spark statement.')
-        c.argument('kind', help='The kind of spark statement.')
+    with self.argument_context('synapse spark statement invoke') as c:
+        c.argument('code', completer=FilesCompleter(),
+                   help='The code of Spark statement. This is either the code contents or use `@<file path>` to load the content from a file')
+        c.argument('language', arg_type=get_enum_type(SparkStatementLanguage), validator=validate_statement_language, help='The language of Spark statement.')
 
     # synapse workspace
-    for scope in ['synapse workspace', 'synapse spark pool', 'synapse sql pool']:
+    for scope in ['synapse workspace', 'synapse spark pool', 'synapse sql pool', 'synapse workspace firewall-rule']:
         with self.argument_context(scope) as c:
-            c.argument('resource_group_name', arg_type=resource_group_name_type, help='The resource group name.')
+            c.argument('resource_group_name', arg_type=resource_group_name_type, id_part='resource_group')

```

Comment:
No need to add "resource_group_name" as it's a global argument that will be added automatically by cli core.

Reply:
removed.

---
In reply to: [424965070](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r424965070) [](ancestors = 424965070)